gateway-service-impl/src/main/java/org/hypertrace/gateway/service/explore/RequestHandler.java

```diff
@@ -114,6 +141,92 @@ QueryRequest buildQueryRequest(
     return builder.build();
   }
 
+  private Optional<List<String>> getEntityIdsToFilter(
```

Comment:
Can we add a comment, on what the function is doing, and how it can be optimised by pre-fetching results from Pinot, instead of 3 queries?